rclcpp/include/rclcpp/strategies/static_memory_strategy.hpp

```diff
@@ -0,0 +1,163 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP_RCLCPP_STATIC_MEMORY_STRATEGY_HPP_
+#define RCLCPP_RCLCPP_STATIC_MEMORY_STRATEGY_HPP_
+
+#include <unordered_map>
+
+#include <rclcpp/memory_strategy.hpp>
+
+namespace rclcpp
+{
+
+namespace memory_strategies
+{
+
+namespace static_memory_strategy
+{
+
+class StaticMemoryStrategy : public memory_strategy::MemoryStrategy
+{
+
+public:
+  StaticMemoryStrategy()
+  {
+    memset(_memory_pool, 0, _pool_size);
+    memset(_subscriber_pool, 0, _max_subscribers);
+    memset(_service_pool, 0, _max_services);
+    memset(_client_pool, 0, _max_clients);
+    memset(_guard_condition_pool, 0, _max_guard_conditions);
+    _pool_seq = 0;
+    _exec_seq = 0;
+
+    // Reserve _pool_size buckets in the memory map.
+    _memory_map.reserve(_pool_size);
+    for (size_t i = 0; i < _pool_size; ++i) {
+      _memory_map[_memory_pool[i]] = 0;
+    }
+
+    for (size_t i = 0; i < _max_executables; ++i) {
+      _executable_pool[i] = std::make_shared<executor::AnyExecutable>(executor::AnyExecutable());
+    }
+  }
+
+  void ** borrow_handles(HandleType type, size_t number_of_handles)
+  {
+    switch (type) {
+      case HandleType::subscriber_handle:
+        return _subscriber_pool;
+      case HandleType::service_handle:
+        return _service_pool;
+      case HandleType::client_handle:
+        return _client_pool;
+      case HandleType::guard_condition_handle:
+        return _guard_condition_pool;
+      default:
+        break;
+    }
+    throw std::runtime_error("Unrecognized enum, could not borrow handle memory.");
+  }
+
+  void return_handles(HandleType type, void ** handles)
+  {
+    switch (type) {
+      case HandleType::subscriber_handle:
+        memset(_subscriber_pool, 0, _max_subscribers);
+        break;
+      case HandleType::service_handle:
+        memset(_service_pool, 0, _max_services);
+        break;
+      case HandleType::client_handle:
+        memset(_client_pool, 0, _max_clients);
+        break;
+      case HandleType::guard_condition_handle:
+        memset(_guard_condition_pool, 0, _max_guard_conditions);
+        break;
+      default:
+        throw std::runtime_error("Unrecognized enum, could not return handle memory.");
+        break;
```

Comment:
To be clear, he means the `break` not the whole `default` case. (It wasn't immediately obvious to me at least)